src/CreatorToken.sol

```diff
@@ -99,17 +99,23 @@ contract CreatorToken is ERC721 {
   }
 
   function buy(uint256 _maxPayment) public {
-    _buy(msg.sender, _maxPayment);
+    uint256 _totalPrice = _buy(msg.sender);
+    if (_totalPrice > _maxPayment) {
+      revert CreatorToken__MaxPaymentExceeded(_totalPrice, _maxPayment);
+    }
   }
 
   function buy(address _to, uint256 _maxPayment) public {
-    _buy(_to, _maxPayment);
+    uint256 _totalPrice = _buy(_to);
+    if (_totalPrice > _maxPayment) {
+      revert CreatorToken__MaxPaymentExceeded(_totalPrice, _maxPayment);
+    }
   }
 
   function bulkBuy(uint256 _numOfTokens, uint256 _maxPayment) public {
```

Comment:
Instead of repeating all the logic, this method should just curry to `bulkBuy` with arrity 3

Reply:
Updated in the same commit as above: [43728e6](https://github.com/showtime-xyz/creator-tokens/pull/29/commits/43728e6415b85c2b58014ab0e5adabb84a43efa5)